# **TASK 1**

In [ ]:
import numpy as np

0.05x + 0.07y + 0z = 2250

0.05x + 0y + 0.06z = 1400

x + y + z = 50000

In [8]:
A = np.array([[0.05, 0.07, 0], [0.05, 0, 0.06], [1, 1, 1]])

In [9]:
A

array([[0.05, 0.07, 0.  ],
       [0.05, 0.  , 0.06],
       [1.  , 1.  , 1.  ]])

In [11]:
B = np.array([2250, 1400, 50000], dtype=float)

In [12]:
B

array([ 2250.,  1400., 50000.])

In [15]:
np.linalg.solve(A,B)

array([10000., 25000., 15000.])

# TASK 2

x + y + z = 1328

x - y = - 120

x - z = 100

In [16]:
A = np.array([[1,1,1],[1,-1,0],[1,0,-1]])

In [17]:
A

array([[ 1,  1,  1],
       [ 1, -1,  0],
       [ 1,  0, -1]])

In [18]:
b = np.array([1328,-120,100])

In [19]:
b

array([1328, -120,  100])

In [20]:
np.linalg.solve(A,b)

array([436., 556., 336.])

Solution:

- 436 items of iphone 6,

- 556 items of iphone 11,

- 336 items of iphone 12

# TASK 3

Пусть x1 = 1/a^2; x2 = 1/b^2; x3 = 1/c^2; тогда составим систему линенйных уравнений, подставив значения координат точек:

3x1 + 0x2 + 3*x3 = 1

6x1 + 0.25x2 + 0*x3 = 1

1x1 + 0.33x2 + 1*x3 = 1

In [21]:
A = np.array([[3,0,3],[6,0.25,0],[1,0.33,1]])

In [22]:
A

array([[3.  , 0.  , 3.  ],
       [6.  , 0.25, 0.  ],
       [1.  , 0.33, 1.  ]])

In [23]:
b = np.array([1,1,1])

In [24]:
b

array([1, 1, 1])

In [25]:
soln = np.linalg.solve(A,b)

In [27]:
soln

array([0.08249158, 2.02020202, 0.25084175])

In [28]:
a2 = 1 / soln[0]
b2 = 1 / soln[1]
c2 = 1 / soln[2]

In [29]:
a2, b2, c2

(12.122448979591837, 0.495, 3.9865771812080535)

# Task 4

a + b + c = 12

9a + 3b + c = 54

a - b + c = 2

In [30]:
A = np.array([[1,1,1],[9,3,1],[1,-1,1]])

In [31]:
A

array([[ 1,  1,  1],
       [ 9,  3,  1],
       [ 1, -1,  1]])

In [32]:
b = np.array([12,54,2])

In [33]:
b

array([12, 54,  2])

In [34]:
results = np.linalg.solve(A,b)
results

array([4., 5., 3.])

Answer: y = 4x^2 + 5x + 3

# Task 5

In [35]:
def get_polynom(coords):
    if len(coords) <= 2:
        return "more than 2 dots are needed"
    if len(coords) > 2:
        x = []
        y = []
        for i in coords:
            x.append(i[0])
            y.append(i[1])
        p = np.poly1d(np.polyfit(x, y, len(coords)-1))
        return p

In [36]:
coords = ((1,0), (3,2), (4,3), (7,1), (9, -3))
print(get_polynom(coords))

         4          3         2
0.01181 x - 0.2465 x + 1.441 x - 2.031 x + 0.825
